omx/gstomxaacdec.c

```diff
@@ -96,70 +98,113 @@ gst_omx_aac_dec_set_format (GstOMXAudioDec * dec, GstOMXPort * port,
   const gchar *stream_format;
 
   gst_omx_port_get_port_definition (port, &port_def);
-  port_def.format.audio.eEncoding = OMX_AUDIO_CodingAAC;
+
+  if (port->index == OMX_AACDEC_INPUT_PORT) {
+    port_def.nPortIndex = port->index;
+    port_def.eDir = OMX_DirInput;
+    port_def.nBufferCountActual = 1;
+    port_def.nBufferCountMin = 1;
+    port_def.nBufferSize = 4096;
+    port_def.bEnabled = OMX_TRUE;
+    port_def.bPopulated = OMX_FALSE;
+    port_def.eDomain = OMX_PortDomainAudio;
+    port_def.bBuffersContiguous = OMX_FALSE;
+    port_def.nBufferAlignment = 32;
+    port_def.format.audio.cMIMEType = (OMX_STRING)"ADEC";
+    port_def.format.audio.pNativeRender = NULL;
+    port_def.format.audio.eEncoding = OMX_AUDIO_CodingAAC;
+    port_def.format.audio.bFlagErrorConcealment = OMX_FALSE;
+    GST_DEBUG_OBJECT (self, "Updating input port definition");
+
+  } else if (port->index == OMX_AACDEC_OUTPUT_PORT) {
+    port_def.nPortIndex = port->index;
+    port_def.eDir = OMX_DirOutput;
+    port_def.nBufferCountActual = 1;
+    port_def.nBufferCountMin = 1;
+    port_def.nBufferSize = 4608; // port_def.nBufferSize = 8192 // TI setup
```

Comment:
Why 4608 and not 8192? Maybe the comment is confusing.

Reply:
66cee90a7d9904dfc8155591da324492d4f520ef